src/ssm.c

```diff
@@ -113,21 +113,36 @@ static void Check_Error(BMS_INPUT_T *input, BMS_STATE_T *state, BMS_OUTPUT_T *ou
     // }
     //[TODO] change to case statement
     if (state->curr_mode != BMS_SSM_MODE_INIT) {
-        int16_t max_cell_temp_thres_C = state->pack_config->max_cell_temp_dC;
```

Comment:
This section is a big one. Basically, the EVT code as is would have not reset the error in the case of a very brief observed error (e.g. 10ms of under voltage) and would have failed once the timeout concluded, even if the error ceased to be present (e.g. voltage went back up). It would also not have recorded multiple errors, but rather only one. This change fixes both problems (or at least what FSAE views as problems).